modules/sdk-coin-polygon/src/polygon.ts

```diff
@@ -306,4 +320,186 @@ export class Polygon extends Eth {
 
     return { halfSigned: txParams };
   }
+
+  /**
+   * Make a query to Etherscan for information such as balance, token balance, solidity calls
+   * @param query {Object} key-value pairs of parameters to append after /api
+   * @returns {Object} response from Etherscan
+   */
+  async recoveryBlockchainExplorerQuery(query: Record<string, string>): Promise<any> {
+    const token = common.Environments[this.bitgo.getEnv()].polygonscanApiToken;
+    if (token) {
+      query.apikey = token;
+    }
+    const response = await request.get(common.Environments[this.bitgo.getEnv()].polygonBaseUrl + '/api').query(query);
+
+    if (!response.ok) {
+      throw new Error('could not reach Polygonscan');
+    }
+
+    if (response.body.status === '0' && response.body.message === 'NOTOK') {
+      throw new Error('Polygonscan rate limit reached');
+    }
+    return response.body;
+  }
+
+  /**
+   * Builds a funds recovery transaction without BitGo
+   * @param params
+   * @param params.userKey {String} [encrypted] xprv or xpub
+   * @param params.backupKey {String} [encrypted] xprv or xpub if the xprv is held by a KRS provider
+   * @param params.walletPassphrase {String} used to decrypt userKey and backupKey
+   * @param params.walletContractAddress {String} the Polygon address of the wallet contract
+   * @param params.krsProvider {String} necessary if backup key is held by KRS
+   * @param params.recoveryDestination {String} target address to send recovered funds to
```

Comment:
Please follow the format @param {String} params.userKey - [encrypted] xprv or xpub